lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts

```diff
@@ -452,9 +475,52 @@ export default class ArcGisMapServerCatalogItem extends UrlMixin(
         return;
       }
 
+      function windowDurationInMs(
+        windowDuration: number | undefined,
+        timeWindowUnit: string | undefined
+      ): number | undefined {
+        if (
+          windowDuration === undefined ||
+          windowDuration === 0 ||
+          timeWindowUnit === undefined
+        ) {
+          return undefined;
+        }
+
+        const rawTimeWindowData: any = {};
+        rawTimeWindowData[timeWindowUnit] = windowDuration;
+        const duration = moment.duration(rawTimeWindowData).asMilliseconds();
+        if (duration === 0) {
+          return undefined;
+        } else {
+          return duration;
+        }
+      }
+
+      function getTimeWindowQueryString(
+        currentTime: number,
+        windowDuration: number
+      ) {
+        const timeBound = Number(currentTime) + windowDuration;
+        if (windowDuration >= 0) {
+          return currentTime + "," + timeBound;
+        } else {
+          return "" + timeBound + "," + currentTime;
+        }
+      }
```

Comment:
Could you move these functions outside the class but in the same file? Should make the method easier to read.